gen2-palm-detection/main.py

```diff
@@ -0,0 +1,468 @@
+# coding=utf-8
+import argparse
+from pathlib import Path
+
+import cv2
+import depthai
+import numpy as np
+from imutils.video import FPS
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-nd", "--no-debug", action="store_true", help="prevent debug output"
+)
+parser.add_argument(
+    "-cam",
+    "--camera",
+    action="store_true",
+    help="Use DepthAI 4K RGB camera for inference (conflicts with -vid)",
+)
+
+parser.add_argument(
+    "-vid",
+    "--video",
+    type=str,
+    help="The path of the video file used for inference (conflicts with -cam)",
+)
+
+args = parser.parse_args()
+
+debug = not args.no_debug
+
+if args.camera and args.video:
+    raise ValueError(
+        'Command line parameter error! "-Cam" cannot be used together with "-vid"!'
+    )
+elif args.camera is False and args.video is None:
+    raise ValueError(
+        'Missing inference source! Use "-cam" to run on DepthAI cameras, or use "-vid <path>" to run on video files'
+    )
+
+
+def to_planar(arr: np.ndarray, shape: tuple):
+    return cv2.resize(arr, shape).transpose((2, 0, 1)).flatten()
+
+
+def to_nn_result(nn_data):
+    return np.array(nn_data.getFirstLayerFp16())
+
+
+def to_tensor_result(packet):
+    return {
+        name: np.array(packet.getLayerFp16(name))
+        for name in [tensor.name for tensor in packet.getRaw().tensors]
+    }
+
+
+def frame_norm(frame, *xy_vals):
+    """
+    nn data, being the bounding box locations, are in <0..1> range -
+    they need to be normalized with frame width/height
+
+    :param frame:
+    :param xy_vals: the bounding box locations
+    :return:
+    """
+    return (
+        np.clip(np.array(xy_vals), 0, 1)
+        * np.array(frame.shape[:2] * (len(xy_vals) // 2))[::-1]
+    ).astype(int)
+
+
+def run_nn(x_in, x_out, in_dict):
+    nn_data = depthai.NNData()
+    for key in in_dict:
+        nn_data.setLayer(key, in_dict[key])
+    x_in.send(nn_data)
+    return x_out.tryGet()
+
+
+class DepthAI:
+    def __init__(
+        self,
+        file=None,
+        camera=False,
+    ):
+        print("Loading pipeline...")
+        self.file = file
+        self.camera = camera
+        self.fps_cam = FPS()
+        self.fps_nn = FPS()
+        self.create_pipeline()
+        self.start_pipeline()
+        self.fontScale = 1 if self.camera else 2
+        self.lineType = 0 if self.camera else 3
+
+    def create_pipeline(self):
+        print("Creating pipeline...")
+        self.pipeline = depthai.Pipeline()
+
+        if self.camera:
+            # ColorCamera
+            print("Creating Color Camera...")
+            self.cam = self.pipeline.createColorCamera()
+            self.cam.setPreviewSize(self._cam_size[1], self._cam_size[0])
+            self.cam.setResolution(
+                depthai.ColorCameraProperties.SensorResolution.THE_4_K
```

Comment:
1080p is enough here
```suggestion
                depthai.ColorCameraProperties.SensorResolution.THE_1080_P
```